src/mewbot/io/rss.py

```diff
@@ -0,0 +1,514 @@
+            entry_link = LINK_NOT_SET_STR
+
+        return str(entry_link)
+
+    @staticmethod
+    def _extract_description(
+        entry: feedparser.util.FeedParserDict, problem_list: List[str]
+    ) -> str:
+        """
+        Extract and return the link from an entry.
+        Adding any problems to the problem list.
+        """
+        # description
+        try:
+            entry_description = entry.description
+        except AttributeError:
+            problem_list.append("entry did not possess a description attribute")
+            entry_description = DESCRIPTION_NOT_SET_STR
+
+        return str(entry_description)
+
+    @staticmethod
+    def _extract_author(
+        entry: feedparser.util.FeedParserDict, problem_list: List[str]
+    ) -> str:
+        try:
+            entry_author = entry.author
+        except AttributeError:
+            problem_list.append("entry did not possess an author attribute")
+            entry_author = AUTHOR_NOT_SET_STR
+
+        return str(entry_author)
+
+    @staticmethod
+    def _extract_category(
+        entry: feedparser.util.FeedParserDict, problem_list: List[str]
+    ) -> str:
+        try:
+            entry_category = entry.category
+        except AttributeError:
+            problem_list.append("entry did not possess a category attribute")
+            entry_category = CATEGORY_NOT_SET_STR
+
+        return str(entry_category)
+
+    @staticmethod
+    def _extract_comments(
+        entry: feedparser.util.FeedParserDict, problem_list: List[str]
+    ) -> str:
+        try:
+            entry_comments = entry.comments
+        except AttributeError:
+            problem_list.append("entry did not possess a comments attribute")
+            entry_comments = COMMENTS_NOT_SET_STR
+
+        return str(entry_comments)
+
+    @staticmethod
+    def _extract_enclosure(entry: feedparser.util.FeedParserDict) -> str:
+        try:
+            entry_enclosure = entry.enclosure
+        except AttributeError:
+            # No logging as this is an optional attribute
+            entry_enclosure = ENCLOSURE_NOT_SET_STR
+
+        return str(entry_enclosure)
+
+    def _extract_guid(
+        self, entry: feedparser.util.FeedParserDict, site_url: str, problem_list: List[str]
+    ) -> str:
+
+        entry_title = self._extract_title(entry=entry, problem_list=problem_list)
+
+        # guid - there are no rules for this syntax
+        # THUS ONLY THE COMBINATION OF GUID AND SITE_URL SHOULD BE ASSUMED UNIQUE
+        # This combination is used to determine if events have already been sent
+        try:
+            entry_guid = entry.guid
+        except AttributeError:
+            problem_list.append(
+                f"entry did not possess a guid - "
+                f"{site_url} might be producing malformed RSS entries"
+            )
+
+            # This is not a fantastic guid - but one needs to be set for internal tracking
+            entry_guid = f"{site_url}:{entry_title}"
```

Comment:
Minor: I'd be more included to go with link and date

Reply:
That does, indeed, make more sense. Corrected.